src/string.c

```diff
@@ -88,46 +88,43 @@ static val_t s_index_of(struct v7 *v7, val_t this_obj, val_t args, int last) {
   val_t s = to_string(v7, this_obj);
   val_t arg0 = v7_array_at(v7, args, 0);
   val_t arg1 = i_value_of(v7, v7_array_at(v7, args, 1));
-  val_t sub, res = v7_create_number(-1);
-  size_t i, n1, n2, fromIndex;
-  const char *p1, *p2;
-
-  if (v7_is_undefined(arg0)) return res;
-
-  if (v7_is_double(arg1)) {
-    double d = v7_to_double(arg1);
-    if (isinf(d) && d > 0) {
-      return v7_create_number(-1);
-    }
-    fromIndex = isnan(d) || isinf(d) ? 0 : d;
-  } else {
-    fromIndex = 0;
-  }
-
-  sub = to_string(v7, arg0);
-  p1 = v7_to_string(v7, &s, &n1);
-  p2 = v7_to_string(v7, &sub, &n2);
-
-  if (n2 > n1) return res;
-
-  if (last) {
-    for (i = n1 - n2; i >= fromIndex; i--) {
-      if (memcmp(p1 + i, p2, n2) == 0) return v7_create_number(i);
+  val_t sub;
+  size_t i, n1, n2, fromIndex = 0;
+  double res = -1;
+  const char *p1, *p2, *end;
+
+  if (arg0 != V7_UNDEFINED) {
```

Comment:
We got rid of direct comparison to V7_UNDEFINED in favour of always usign v7_is_undefined (and v7_create_undefined). 

The rationale is that soon we'll replace `val_t` definition with a `union { double d; }`